homeassistant/components/homekit/util.py

```diff
@@ -34,10 +40,47 @@ def validate_entity_config(values):
             code = config.get(ATTR_CODE)
             params[ATTR_CODE] = cv.string(code) if code else None
 
+        if domain == 'media_player':
+            mode = config.get(CONF_MODE)
+            if mode:
```

Comment:
We don't need this check. `cv.ensure_list(value)` returns an empty list if the `value` is `None`. Additionally I think changing the `for` loop `mode` to `key` might be a bit better.